src/pages/BookingDetails/BookingDetails.tsx

```diff
@@ -0,0 +1,463 @@
+
+  const navigate = useNavigate();
+
+  let total_price = totalPrice;
+
+  const { data, loading }: ResAddon = UseFetch(`addon`);
+  let addOns = {};
+  if (!loading) {
+    data.forEach((item) => {
+      addOns[item.name] = Number(item.price);
+    });
+  }
+
+  const [selectCheck, setSelectCheck] = useState({
+    addon: [],
+  });
+
+  const [total, setTotal] = useState(0);
+  const [discount, setDiscount] = useState(0);
+
+  useEffect(() => {
+    Object.values(filtAddOn()).forEach((each: any) => {
+      total_price += each;
+    });
+    setTotal(
+      total_price +
+        addOnPrice() -
+        (total_price + addOnPrice()) * (discount / 100)
+    );
+  }, [selectCheck]);
+
+  const handleAddon = (e) => {
+    if (e.target.checked) {
+      setSelectCheck({
+        addon: [...selectCheck.addon, e.target.value],
+      });
+    }
+    if (!e.target.checked) {
+      setSelectCheck({
+        addon: selectCheck.addon.filter((item) => item !== e.target.value),
+      });
+    }
+  };
+
+  const filtAddOn = () => {
+    let a = {};
+    selectCheck.addon.forEach((each) => {
+      if (Object.keys(addOns).includes(each)) {
+        a[each] = addOns[each];
+      }
+    });
+    return a;
+  };
+
+  const handleCoupon = async (e) => {
+    e.preventDefault();
+    // const coupon = couponRef.current?.value;
+    const res = await axios.post("coupon/validate", {
+      coupon,
+    });
+
+    if (res.data == "Invalid Coupon") {
+      toast.error(res.data);
+    } else {
+      setDiscount(res.data);
+      toast.success("Coupon Applied Successfully");
+      setApply(true);
+      couponRef.current.disabled = true;
+      let price = total * (res.data / 100);
+      setTotal(total - price);
+      buttonRef.current.disabled = true;
+    }
+  };
+
+  const handleRemoveCoupon = async (e) => {
+    e.preventDefault();
+    const res = await axios.post("coupon/validate", {
+      coupon,
+    });
+    setDiscount(0);
+    toast.success("Coupon Removed Successfully");
+    couponRef.current.disabled = false;
+    setTotal(totalPrice);
+    buttonRemRef.current.disabled = true;
+  };
+
+  const addOnPrice = () => {
+    let price = 0;
+    Object.values(filtAddOn()).forEach((each: any) => {
+      price += each;
+    });
+    return price;
+  };
+
+  const handleSubmit = async (e: any) => {
+    e.preventDefault();
+    const name = nameRef.current?.value;
+    const email = emailRef.current?.value;
+    const specialReq = specialReqRef.current?.value;
+
+    if (name === "" || email === "") {
+      toast.error("Please fill all the fields");
+      return;
+    }
+
+    const selectedAddons = filtAddOn();
+    const data = {
+      name,
+      email,
+      date: new Date(),
+      specialReq,
+      selectedAddons,
+      no,
+      checkin,
+      checkout,
+      roomType,
+      roomPrice,
+      coupon: {
+        coupon,
+        discount: discount.toString(),
+      },
+      total:
+        total_price +
```

Comment:
I see the code for calculating the `totalprice `is repeated many times worth factoring it out